features/profiles/utils/formatUsername.ts

```diff
@@ -0,0 +1,23 @@
+import { getConfig } from "@/config";
+
+/**
+ * Formats usernames from Converse domains by extracting the username part and adding @ prefix
+ * Returns undefined for non-Converse usernames
+ * @param username - The username to format
+ * @returns The formatted username with @ prefix if it's a Converse username, undefined otherwise
+ */
+export function formatUsername(
+  username: string | undefined
+): string | undefined {
+  if (!username) return undefined;
+
+  // Check if it's a Converse username (either domain)
+  if (username.endsWith(getConfig().usernameSuffix)) {
+    // Extract everything before the domain
+    const cleanUsername = username.replace(getConfig().usernameSuffix, "");
+    return `@${cleanUsername}`;
+  }
+
+  // Return undefined for non-Converse usernames
+  return undefined;
```

Comment:
if we want to indicate this isn't a converse username, we'd likely want a result type of some sort
result:
isConverseUsername: true
usernameFormatted

isConverseUsername: false
rawUsername